pkg/action/upgrade.go

```diff
@@ -181,6 +181,17 @@ func (u *UpgradeAction) Run() (err error) {
 	}
 	cleanup.Push(func() error { return e.UnmountImage(&upgradeImg) })
 
+	// Create extra dirs in rootfs as afterwards this will be impossible due to RO system
+	for _, d := range u.spec.ExtraDirsRootfs {
+		if exists, _ := fsutils.Exists(u.config.Fs, filepath.Join(upgradeImg.MountPoint, d)); !exists {
+			u.config.Logger.Debugf("Creating extra dir %s under %s", d, upgradeImg.MountPoint)
+			err = u.config.Fs.Mkdir(filepath.Join(upgradeImg.MountPoint, d), constants.DirPerm)
+			if err != nil {
+				u.config.Logger.Warnf("Failure creating extra dir %s in rootfs at %s", d, upgradeImg.MountPoint)
+			}
+		}
+	}
```

Comment:
These three are the same no? can we make it a single function, so we don't have to keep them up to date whenever one changes?

Reply:
Yep, indeed. I got a bit confused due to the upgrade source being different, but in the end it's the same thing, a mountpoint and mkdir.


Will extract it. Although I'm wary of extracting too much stuff into the utils as that causes circular deps...